c7n/resources/s3.py

```diff
@@ -2494,3 +2494,124 @@ def process_bucket(self, bucket):
 
         s3.put_bucket_lifecycle_configuration(
             Bucket=bucket['Name'], LifecycleConfiguration={'Rules': config})
+
+@actions.register('set-bucket-encryption')
+class SetBucketEncryption(BucketActionBase):
+    """Action enables default encryption on S3 buckets
+
+    `crypto`: aws:kms | AES256` Optional: Defaults to AES256
+    `key`: arn, or id kms key
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: s3-enable-default-encryption-kms
+                resource: s3
+                actions:
+                  - type: set-bucket-encryption
+                    key: 1234abcd-12ab-34cd-56ef-1234567890ab
+
+              - name: s3-enable-default-encryption-aes256
+                resource: s3
+                actions:
+                  - type: set-bucket-encryption
+    """
+
+    schema = {
+        'type': 'object',
+        'additionalProperties': False,
+        'properties': {
+            'type': {'enum': ['set-bucket-encryption']},
+            'key': {'type': 'string'}
+        }
+    }
+
+    permissions = ( 's3:PutEncryptionConfiguration',
+                    's3:GetEncryptionConfiguration')
+
+    def process(self, buckets):
+        with self.executor_factory(max_workers=3) as w:
+            futures = {}
+            results = []
+
+            config = { 'Rules': [
+                    {
+                        'ApplyServerSideEncryptionByDefault': {
+                            'SSEAlgorithm': 'AES256'
+                        }
+                    }
+                ]}
+
+            if 'key' in self.data:
+                config['Rules'][0]['ApplyServerSideEncryptionByDefault']['SSEAlgorithm'] = 'aws:kms'
+                config['Rules'][0]['ApplyServerSideEncryptionByDefault']['KMSMasterKeyID'] = self.data['key']
+
+            for b in buckets:
+                futures[w.submit(self.process_bucket, b, config)] = b
+
+            for future in as_completed(futures):
+                if future.exception():
+                    bucket = futures[future]
+                    self.log.error('error enabling bucket encryption: %s\n%s',
+                                   bucket['Name'], future.exception())
+                results += filter(None, [future.result()])
+
+            return results
+
+    def process_bucket(self, bucket, config):
+        s3 = bucket_client(local_session(self.manager.session_factory), bucket)
+        b = bucket['Name']
+        try:
```

Comment:
you are already handling the exception in the as_completed block in the process bucket above.  It's redundant here.